plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -736,26 +737,54 @@ public synchronized void commentTable(String databaseName, String tableName, Opt
     @Override
     public synchronized void commentColumn(String databaseName, String tableName, String columnName, Optional<String> comment)
     {
-        alterTable(databaseName, tableName, oldTable -> {
-            if (oldTable.getColumn(columnName).isEmpty()) {
-                SchemaTableName name = new SchemaTableName(databaseName, tableName);
-                throw new ColumnNotFoundException(name, columnName);
-            }
+        alterTable(databaseName, tableName, table -> {
+            List<Column> dataColumns = table.getDataColumns();
+            List<Column> partitionColumns = table.getPartitionColumns();
 
-            ImmutableList.Builder<Column> newDataColumns = ImmutableList.builder();
-            for (Column fieldSchema : oldTable.getDataColumns()) {
-                if (fieldSchema.getName().equals(columnName)) {
-                    newDataColumns.add(new Column(columnName, fieldSchema.getType(), comment, fieldSchema.getProperties()));
-                }
-                else {
-                    newDataColumns.add(fieldSchema);
-                }
+            Optional<Integer> matchingDataColumn = indexOfColumnWithName(dataColumns, columnName);
+            Optional<Integer> matchingPartitionColumn = indexOfColumnWithName(partitionColumns, columnName);
+
+            if (matchingDataColumn.isPresent() && matchingPartitionColumn.isPresent()) {
```

Comment:
when can this happen?

Reply:
I don't know. I've copied it from Glue meta store. Better safe than sorry.